extra/cache-extra/TokenParser/CacheTokenParser.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extra\Cache\TokenParser;
+
+use Twig\Error\SyntaxError;
+use Twig\Extra\Cache\Node\CacheNode;
+use Twig\Node\Node;
+use Twig\Token;
+use Twig\TokenParser\AbstractTokenParser;
+
+class CacheTokenParser extends AbstractTokenParser
+{
+    public function parse(Token $token): Node
+    {
+        $stream = $this->parser->getStream();
+        $expressionParser = $this->parser->getExpressionParser();
+        $key = $expressionParser->parseExpression();
+
+        $ttl = null;
+        $tags = null;
+        while ($stream->test(Token::NAME_TYPE)) {
+            $k = $stream->getCurrent()->getValue();
+            $stream->next();
+            $args = $expressionParser->parseArguments();
+
+            switch ($k) {
+                case 'ttl':
+                    $ttl = $args->getNode(0);
+                    break;
+                case 'tags':
+                    $tags = $args->getNode(0);
```

Comment:
should we throw a proper SyntaxError if we either have 0 arguments or multiple arguments ? For now, multiple arguments would be ignored silently (not reporting a mistake of writing `tags('foo', 'bar')` instead of `tags(['foo', 'bar'])`) and no arguments is likely to break with an error that's not clear about what the issue is (reporting a bug in the token parser instead)

Reply:
TBH, I was too lazy to do it, but as you noticed it, I will do it :)